packages/shared/src/store/sync.ts

```diff
@@ -393,6 +393,17 @@ export const syncPushNotificationsSetting = async (ctx?: SyncCtx) => {
   await db.setPushNotificationsSetting(setting);
 };
 
+async function handleChannelAdd(channel: db.Channel) {
+  await db.setJoinedGroupChannels({
+    channelIds: [channel.id],
+  });
+  const channelInDb = await db.getChannel({ id: channel.id });
+  // Only insert the channel if the user has permission to see it
+  if (channelInDb?.currentUserIsMember) {
+    await db.insertChannels([channel]);
+  }
+}
+
```

Comment:
I'm trying to understand this one. Is this intended to handle both channel adds and updates? 

If we've got a new channel to add, we first call `setJoinedGroupChannels`, which updates an existing channel in the db, and then we call `getChannel`, before inserting the channel to the db? It seems like this would always fail for inserting new channels, since those first checks will fail.

If it's an update, why are we calling insertChannels rather than updateChannel?

Reply:
I also think we should handle the permissions checks within the `updateChannel` query to avoid potential race conditions.

Reply:
I was attempting to insert all new channels through a permission check. I'll ping you when I get to a desktop, but I'll take a look at your suggestion first.

Reply:
Alright-take a look again when you get a sec. I put everything for when we hear about during addChannel and updateChannel in queries (as well as the updateChannel action itself) through a new function (`upsertChannel`), which should avoid any inconsistencies. Same behavior as before.

We still use `insertChannels` elsewhere (for DM invites, creating group DMs, etc). Not sure if we want to clean that function up or consolidate, happy to hear thoughts.